hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableAirdropStoreImpl.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl;
+
+import static com.hedera.node.app.service.token.impl.schemas.V0500TokenSchema.AIRDROPS_KEY;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.PendingAirdropId;
+import com.hedera.hapi.node.base.PendingAirdropValue;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.service.token.ReadableAirdropStore;
+import com.swirlds.state.spi.ReadableKVState;
+import com.swirlds.state.spi.ReadableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Default implementation of {@link ReadableAccountStore}
+ */
+public class ReadableAirdropStoreImpl implements ReadableAirdropStore {
+    /** The underlying data storage class that holds the airdrop data. */
+    private final ReadableKVState<PendingAirdropId, PendingAirdropValue> readableAirdropState;
+
+    /**
+     * Create a new {@link ReadableAirdropStoreImpl} instance.
+     *
+     * @param states The state to use.
+     */
+    public ReadableAirdropStoreImpl(@NonNull final ReadableStates states) {
+        requireNonNull(states);
+        this.readableAirdropState = states.get(AIRDROPS_KEY);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean exists(@NonNull final PendingAirdropId airdropId) {
+        return readableAirdropState.contains(requireNonNull(airdropId));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    @Nullable
+    public PendingAirdropValue getFungibleAirdropAmount(@NonNull final PendingAirdropId airdropId) {
+        requireNonNull(airdropId);
+        if (airdropId.hasNonFungibleToken()) {
+            return null;
+        }
```

Comment:
Why do we need this ? We should just return `readableAirdropState.get(airdropId)` and this will be null, if there is no value